lib/base-formatter.js

```diff
@@ -29,23 +29,30 @@ export class BaseFormatter {
         this.formatterInfo = formatterInfo;
     }
 
-    async format(args, source) {
+    /**
+     * Format the provided source code using the formatting tool.
+     *
+     * This function should construct the final arguments passed to the tool
+     * executable.
+     *
+     * Optionally accept a third argument to respect other formatting options.
+     * The possible options passed are as follows:
+     *
+     * interface AdditionalFormatOptions {
+     *     useSpaces: boolean;
+     *     tabWidth: number;
+     *     baseStyle: string;
+     * }
+     */
+    async format(source, options) {
+        const args = [`--style=${options.baseStyle}`];
         return await exec.execute(this.formatterInfo.exe, args, {input: source});
     }
 
-    getDefaultArguments() {
-        return [];
-    }
-
-    getStyleArguments(style) {
-        return [`--style=${style}`];
-    }
-
+    /**
+     * Test if a formatting base style is valid for this formatter
+     */
     isValidStyle(style) {
         return this.formatterInfo.styles.includes(style);
     }
-
-    isOneTrueStyle() {
-        return this.formatterInfo.styles.length === 0;
-    }
```

Comment:
The formatter itself no longer cares if it is a "one true style" formatter. Implementor can express this in `isValidStyle`, and should handle this appropriately in `format`